java/memory/memory-core/src/main/java/org/apache/arrow/memory/AllocationOutcome.java

```diff
@@ -19,15 +19,18 @@
 
 import java.util.Optional;
 
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+
 /**
  * Describes the type of outcome that occurred when trying to account for allocation of memory.
  */
 public class AllocationOutcome {
   private final Status status;
```

Comment:
@davisusanibar should we annotate `Status`? Seems like it is `NonNullable`?

Reply:
Yes, by default CheckerFramewrok assume all variables as [@NonNull](https://checkerframework.org/api/org/checkerframework/checker/nullness/qual/NonNull.html#:~:text=This%20annotation%20is%20rarely%20written%20in%20source%20code%2C%20because%20it%20is%20the%20default.).